networkx/classes/digraph.py

```diff
@@ -266,6 +285,10 @@ class DiGraph(Graph):
     a dictionary-like object.
     """
 
+    _adj = _CachedPropertyResetterAdjAndSucc()
+    _succ = _CachedPropertyResetterAdjAndSucc()
```

Comment:
Realistically I think the best way to get mypy to not complain is just to mark the differences with:

```suggestion
    _adj = _CachedPropertyResetterAdjAndSucc()  # type: ignore
    _succ = _CachedPropertyResetterAdjAndSucc()  # type: ignore
```

I found [this SO post](https://stackoverflow.com/questions/58377106/why-does-mypy-report-incompatible-types-when-child-and-parent-class-both-satisfy) which explains what's happening. This is fine and makes sense *in the case where types are explicitly defined*. IMO it's a bit annoying mypy is inferring typing info from the underlying objects in the cases where we don't provide specific typing information (if we were, I assume we'd be able to get around this error by adding an annotation that allows either of the various `_CachedProperty*` objects).

Either way, I think the best thing to do is just ignore it for now!